sympy/stats/stochastic_process_types.py

```diff
@@ -638,11 +708,34 @@ def transition_probabilities(self):
         """
         return self.args[2]
 
+    @property
+    def num_states(self):
+        """
+        The number of states in the Markov Chain. Can be symbolic.
+        """
+        if self.transition_probabilities is None:
+            return None
+        else:
+            return self.transition_probabilities.shape[0]
+
+    @property
+    def is_irreducible(self):
+        """
+        A Markov Chain is irreducible iff it consists of exactly
+        one communication class.
+        """
+        if self._is_numeric:
+            return len(self.communication_classes()) == 1
+        else:
+            return None  # cannot be known
+
     def _transient2transient(self):
         """
         Computes the one step probabilities of transient
         states to transient states. Used in finding
-        fundamental matrix, absorbing probabilties.
+        fundamental matrix, absorbing probabilities.
+
+        This will be depreciated in the future.
```

Comment:
Why?

Reply:
This is a private method that's not used any more. These methods were causing the main problems that point number 1 fixes.